### PR TITLE
Doc: Improve mclock config reference documentation & update PendingReleaseNotes.

### PR DESCRIPTION
**Pull request by sseshasa**
Improve the documentation around.
 - mclock client types.
 - Describe in greater detail about mclock config profiles.
 - Add notes about manually benchmarking OSDs and tuning bluestore throttle
    parameters
 - Include a couple of missing configuration options.
 - Update PendingReleaseNotes with link to mclock reference document.

Fixes: https://tracker.ceph.com/issues/54619
Signed-off-by: Sridhar Seshasayee <sseshasa@redhat.com>

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by sseshasa** [comment]
jenkins render docs

**Comment by sseshasa** [comment]
jenkins test docs

**Comment by sseshasa** [comment]
jenkins retest this please

File: doc/rados/configuration/mclock-config-ref.rst [hunk 1]
```diff
@@ -4,22 +4,52 @@
...
+operations.
+
+
+.. index:: mclock; mclock clients
+
+mClock Client Types
+===================
+
+Depending on the type of requests handled, mclock clients are classified into
+the buckets as shown in the table below,
```
Comment (neha-ojha) [inline doc/rados/configuration/mclock-config-ref.rst:38]: This sentence can be a bit confusing. Can we clarify what we mean by mclock clients here?

File: doc/rados/configuration/mclock-config-ref.rst [hunk 2]
```diff
@@ -48,32 +77,74 @@ described in the :ref:`dmclock-qos` section.
...
+Built-in Profiles
+-----------------
+Users can choose between the following built-in profile types:
+
+high_client_ops (*default*)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+This profile allocates more reservation and limit to external-client ops
+as compared to background recoveries and other internal clients within
+Ceph. This profile is enabled by default. The table shows the resource
+controls parameters set by the profile:
```
Comment (neha-ojha) [inline doc/rados/configuration/mclock-config-ref.rst:91]: nit: control

File: doc/rados/configuration/mclock-config-ref.rst [hunk 3]
```diff
@@ -4,22 +4,52 @@
...
+|  Client Type           | Request Types                                      |
++========================+====================================================+
+| Client                 | IO requests issued by external clients of Ceph     |
++------------------------+----------------------------------------------------+
+| Background recovery    | Internal recovery/backfill requests                |
++------------------------+----------------------------------------------------+
+| Background best-effort | Internal scrub, snap trim and PG deletion requests |
++------------------------+----------------------------------------------------+
+
+The mclock profiles allocate parameters like reservation, weight and limit
```
Comment (neha-ojha) [inline doc/rados/configuration/mclock-config-ref.rst:50]: maybe link https://ceph--45522.org.readthedocs.build/en/45522/rados/configuration/osd-config-ref/#core-concepts to explain what reservation, weight and limit are

File: doc/rados/configuration/mclock-config-ref.rst [hunk 4]
```diff
@@ -48,32 +77,74 @@ described in the :ref:`dmclock-qos` section.
...
+mclock profiles can be broadly classified into *built-in* and *custom* profiles,
+
+Built-in Profiles
+-----------------
+Users can choose between the following built-in profile types:
+
+high_client_ops (*default*)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+This profile allocates more reservation and limit to external-client ops
+as compared to background recoveries and other internal clients within
```
Comment (neha-ojha) [inline doc/rados/configuration/mclock-config-ref.rst:89]: This profile optimizes client performance over background activities, by allocating more reservation and limit to client operations compared to background operations in the OSD.

File: doc/rados/configuration/mclock-config-ref.rst [hunk 5]
```diff
@@ -48,32 +77,74 @@ described in the :ref:`dmclock-qos` section.
...
++------------------------+-------------+--------+-------+
+| background recovery    | 25%         | 1      | 100%  |
++------------------------+-------------+--------+-------+
+| background best-effort | 25%         | 1      | MAX   |
++------------------------+-------------+--------+-------+
+
+high_recovery_ops
+^^^^^^^^^^^^^^^^^
+This profile allocates more reservation to background recoveries as
+compared to external clients and other internal clients within Ceph. For
```
Comment (neha-ojha) [inline doc/rados/configuration/mclock-config-ref.rst:106]: the use of "internal clients" can be confusing to the end user, how about "other background operations within the OSD"?

File: doc/rados/configuration/mclock-config-ref.rst [hunk 6]
```diff
@@ -48,32 +77,74 @@ described in the :ref:`dmclock-qos` section.
...
+example, an admin may enable this profile temporarily to speed-up background
```
Comment (neha-ojha) [inline doc/rados/configuration/mclock-config-ref.rst:107]: nit: administrator

File: doc/rados/configuration/mclock-config-ref.rst [hunk 7]
```diff
@@ -48,32 +77,74 @@ described in the :ref:`dmclock-qos` section.
...
++------------------------+-------------+--------+-------+
+| background recovery    | 60%         | 2      | 200%  |
++------------------------+-------------+--------+-------+
+| background best-effort | 1 (MIN)     | 1      | MAX   |
++------------------------+-------------+--------+-------+
+
+balanced
+^^^^^^^^
+This profile allocates equal reservation to client ops and background
+recovery ops.
```
Comment (neha-ojha) [inline doc/rados/configuration/mclock-config-ref.rst:124]: When should it be used?

File: doc/rados/configuration/mclock-config-ref.rst [hunk 8]
```diff
@@ -48,32 +77,74 @@ described in the :ref:`dmclock-qos` section.
...
++========================+=============+========+=======+
+| client                 | 40%         | 1      | 100%  |
++------------------------+-------------+--------+-------+
+| background recovery    | 40%         | 1      | 150%  |
++------------------------+-------------+--------+-------+
+| background best-effort | 20%         | 1      | MAX   |
++------------------------+-------------+--------+-------+
+
+.. note:: Across the built-in profiles, internal background best-effort clients
+          of mclock (for example "scrub", "snap trim", and "pg deletion") are
```
Comment (neha-ojha) [inline doc/rados/configuration/mclock-config-ref.rst:137]: let's remove "for example", there aren't more

File: doc/rados/configuration/mclock-config-ref.rst [hunk 9]
```diff
@@ -48,32 +77,74 @@ described in the :ref:`dmclock-qos` section.
...
+          given lower reservations but no limits(MAX). This ensures that
+          requests from such clients are able to complete quickly if there are
+          no other competing services.
```
Comment (neha-ojha) [inline doc/rados/configuration/mclock-config-ref.rst:140]: s/services/operations/

File: doc/rados/configuration/mclock-config-ref.rst [hunk 10]
```diff
@@ -48,32 +77,74 @@ described in the :ref:`dmclock-qos` section.
...
+
+
+Custom Profile
+--------------
+This profile gives users complete control over all the mclock
+configuration parameters. Using this profile is not recommended without
+a deep understanding of mclock and related Ceph-configuration options.
```
Comment (neha-ojha) [inline doc/rados/configuration/mclock-config-ref.rst:147]: This profile should be used with caution and it is meant for advanced users, who understand mclock and Ceph related configuration options.

File: doc/rados/configuration/mclock-config-ref.rst [hunk 11]
```diff
@@ -4,22 +4,52 @@
 
 .. index:: mclock; configuration
 
-Mclock profiles mask the low level details from users, making it
-easier for them to configure mclock.
+QoS support in Ceph is implemented using a queuing scheduler based on `the
+dmClock algorithm`_. See :ref:`dmclock-qos` section for more details.
+
+.. note:: The *mclock_scheduler* is supported for Bluestore OSDs. For Filestore
```
[the remark this answers is not in the document]
Comment (sseshasa) [reply doc/rados/configuration/mclock-config-ref.rst]: Done and s/Filestore/FileStore.

File: doc/rados/configuration/mclock-config-ref.rst (hunk 1 above)
Comment (sseshasa) [reply doc/rados/configuration/mclock-config-ref.rst:40]: I have added a couple of sentences to clarify this.

File: doc/rados/configuration/mclock-config-ref.rst (hunk 2 above)
Comment (sseshasa) [reply doc/rados/configuration/mclock-config-ref.rst]: Done.

File: doc/rados/configuration/mclock-config-ref.rst (hunk 3 above)
Comment (sseshasa) [reply doc/rados/configuration/mclock-config-ref.rst:52]: Added the existing reference to the osd-config-ref document.

File: doc/rados/configuration/mclock-config-ref.rst (hunk 4 above)
Comment (sseshasa) [reply doc/rados/configuration/mclock-config-ref.rst]: Done. Thanks for the suggestion.

File: doc/rados/configuration/mclock-config-ref.rst (hunk 6 above)
Comment (sseshasa) [reply doc/rados/configuration/mclock-config-ref.rst]: Done.

File: doc/rados/configuration/mclock-config-ref.rst (hunk 11 above)
[the remark this answers is not in the document]
Comment (anthonyeleven) [reply doc/rados/configuration/mclock-config-ref.rst]: A while back I trolled around and came to the conclusion that the proper forms were `Filestore` and `BlueStore`, with only the latter CamelCased. @neha-ojha do you know for sure?

File: doc/rados/configuration/mclock-config-ref.rst (hunk 7 above)
Comment (sseshasa) [reply doc/rados/configuration/mclock-config-ref.rst]: I added the content for the balanced profile. Please take a look.

File: doc/rados/configuration/mclock-config-ref.rst (hunk 8 above)
Comment (sseshasa) [reply doc/rados/configuration/mclock-config-ref.rst]: Done.

File: doc/rados/configuration/mclock-config-ref.rst (hunk 9 above)
Comment (sseshasa) [reply doc/rados/configuration/mclock-config-ref.rst]: Changed.

File: doc/rados/configuration/mclock-config-ref.rst (hunk 11 above)
[the remark this answers is not in the document]
Comment (sseshasa) [reply doc/rados/configuration/mclock-config-ref.rst]: @anthonyeleven I changed it back to `Filestore` as I think you are correct.

**Comment by sseshasa** [comment]
jenkins test make check

**Review by neha-ojha — APPROVED** [review]

**Pull request merged**